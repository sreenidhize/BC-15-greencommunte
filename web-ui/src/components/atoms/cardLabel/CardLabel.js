import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles({
  primary: {
    fontSize: "18px",
    color: "#324552",
    marginBottom: "8px",
  },
  secondary: {
    fontSize: "14px",
    color: "#5f7381",
    marginBottom: "8px",
  },
});

const CardLabel = ({ label, isPrimary }) => {
  const classes = useStyles();

  const typoStyle = isPrimary ? classes.primary : classes.secondary;
  return (
    <Typography
      className={typoStyle}
      variant="h4"
      data-testid="cardLabel-placeholder"
    >
      {label}
    </Typography>
  );
};

CardLabel.propTypes = {
  label: PropTypes.string,
  isPrimary: PropTypes.bool,
};

export default CardLabel;
