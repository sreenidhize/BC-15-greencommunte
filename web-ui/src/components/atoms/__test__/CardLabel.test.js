import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CardLabel from '../cardLabel/CardLabel';

describe("CardLabel placeholder test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <CardLabel label="Hyderabad" isPrimary />
      );
      expect(container).toMatchSnapshot();
    });
  
    test("test CardLabel render", () => {
      const { getByTestId } = render(<CardLabel label="Hyderabad" isPrimary />);
      expect(getByTestId("cardLabel-placeholder")).toBeInTheDocument();
    });
  });