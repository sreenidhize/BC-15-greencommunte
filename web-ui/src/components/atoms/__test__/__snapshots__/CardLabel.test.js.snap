// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLabel placeholder test Matches the snapshot 1`] = `
<div>
  <h4
    class="MuiTypography-root makeStyles-primary-1 MuiTypography-h4"
    data-testid="cardLabel-placeholder"
  >
    Hyderabad
  </h4>
</div>
`;
